src/plugins/maps_dashboards/public/components/layer_control_panel/layer_control_panel.tsx

```diff
@@ -15,40 +15,127 @@ import {
   const [isLayerConfigVisible, setIsLayerConfigVisible] = useState(false);
   const [isLayerControlVisible, setIsLayerControlVisible] = useState(true);
+
   const [selectedLayerConfig, setSelectedLayerConfig] = useState<ILayerConfig>({
     iconType: '',
-    label: '',
     name: '',
     type: '',
     id: '',
+    zoomRange: [],
+    opacity: 1,
+    visibility: '',
   });
 
-  // TODO: replace it once layers model ready
-  const demoLayers: ILayerConfig[] = [
+  const [layers, setLayers] = useState<ILayerConfig[]>([
     {
-      label: 'Base Map',
       iconType: 'visMapRegion',
       id: 'example_id_1',
-      type: LAYER_TYPE.BASE_MAP,
+      type: MAP_LAYER_TYPE.BASE_MAP,
       name: 'Base Map Layer',
+      zoomRange: [0, 12],
+      opacity: 1,
+      visibility: LAYER_VISIBILITY.VISIBLE,
+      update() {
+        const maplibreInstance = maplibreRef.current;
+        if (maplibreInstance) {
+          const baseMapJson = maplibreInstance.getStyle().layers;
+          if (baseMapJson) {
+            baseMapJson.forEach((mbLayer) => {
+              maplibreInstance.setLayerZoomRange(mbLayer.id, this.zoomRange[0], this.zoomRange[1]);
+              // it will catch error when update opacity in symbol type layer, need figure out later
+              if (mbLayer.type === 'symbol') {
+                return;
+              }
+              maplibreInstance.setPaintProperty(
+                mbLayer.id,
+                `${mbLayer.type}-opacity`,
+                this.opacity
+              );
+              maplibreInstance.setLayoutProperty(mbLayer.id, 'visibility', this.visibility);
+            });
+          } else {
+            maplibreInstance.setStyle(MAP_VECTOR_TILE_URL);
+          }
+        }
+      },
+      hide() {
+        const maplibreInstance = maplibreRef.current;
+        if (maplibreInstance) {
+          const baseMapJson = maplibreInstance.getStyle().layers;
+          if (baseMapJson) {
+            baseMapJson.forEach((mbLayer) => {
+              maplibreInstance.setLayoutProperty(mbLayer.id, 'visibility', this.visibility);
+            });
+          }
+        }
+      },
+      remove() {
+        const maplibreInstance = maplibreRef.current;
+        if (maplibreInstance) {
+          const baseMapJson = maplibreInstance.getStyle().layers;
+          if (baseMapJson) {
+            baseMapJson.forEach((mbLayer) => {
+              maplibreInstance.removeLayer(mbLayer.id);
+            });
+          }
+        }
+      },
     },
-    {
-      label: 'Cluster Layer',
-      iconType: 'visMapRegion',
-      id: 'example_id_2',
-      type: LAYER_TYPE.CLUSTER_MAP,
-      name: 'Cluster Layer',
-    },
-  ];
+  ]);
+
+  const onClickHideLayerButton = () => {
+    if (selectedLayerConfig.visibility === LAYER_VISIBILITY.VISIBLE) {
+      selectedLayerConfig.visibility = LAYER_VISIBILITY.NONE;
+    } else if (selectedLayerConfig.visibility === LAYER_VISIBILITY.NONE) {
```

Comment:
do you need if inside else? can we use else directly?

Reply:
sure, will update.